### PR TITLE
Add unitycatalog-ai tutorials for integrations utilizing UnityCatalog OSS

### PR DESCRIPTION
**Pull request by BenWilson2**
**PR Checklist**

- [X] A description of the changes is added to the description of this PR.
- [ ] If there is a related issue, make sure it is linked to this PR.
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added or modified a feature, documentation in `docs` is updated

**Description of changes**

Adds tutorials for the unitycatalog-ai integrations for interacting with UC OSS

### CODE REVIEW

File: ai/integrations/anthropic/anthropic_oss_example.ipynb [hunk 1]
```diff
@@ -0,0 +1,302 @@
...
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "config = Configuration()\n",
+    "config.host = \"http://localhost:8080/api/2.1/unity-catalog\"\n",
+    "\n",
+    "# The base ApiClient is async\n",
+    "api_client = ApiClient(configuration=config)\n",
+    "\n",
+    "client = UnitycatalogFunctionClient(uc=api_client)\n",
```
Comment (BenWilson2) [inline ai/integrations/anthropic/anthropic_oss_example.ipynb:80]:
```suggestion
    "client = UnitycatalogFunctionClient(api_client=api_client)\n",
```

File: ai/integrations/autogen/autogen_oss_example.ipynb [hunk 2]
```diff
@@ -0,0 +1,498 @@
...
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "config = Configuration()\n",
+    "config.host = \"http://localhost:8080/api/2.1/unity-catalog\"\n",
+    "\n",
+    "# The base ApiClient is async\n",
+    "api_client = ApiClient(configuration=config)\n",
+    "\n",
+    "client = UnitycatalogFunctionClient(uc=api_client)\n",
```
Comment (BenWilson2) [inline ai/integrations/autogen/autogen_oss_example.ipynb:87]:
```suggestion
    "client = UnitycatalogFunctionClient(api_client=api_client)\n",
```

File: ai/integrations/crewai/crewai_oss_example.ipynb [hunk 3]
```diff
@@ -0,0 +1,481 @@
...
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "config = Configuration()\n",
+    "config.host = \"http://localhost:8080/api/2.1/unity-catalog\"\n",
+    "\n",
+    "# The base ApiClient is async\n",
+    "api_client = ApiClient(configuration=config)\n",
+    "\n",
+    "client = UnitycatalogFunctionClient(uc=api_client)\n",
```
Comment (BenWilson2) [inline ai/integrations/crewai/crewai_oss_example.ipynb:79]:
```suggestion
    "client = UnitycatalogFunctionClient(api_client=api_client)\n",
```

File: ai/integrations/langchain/langchain_oss_example.ipynb [hunk 4]
```diff
@@ -0,0 +1,423 @@
...
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "config = Configuration()\n",
+    "config.host = \"http://localhost:8080/api/2.1/unity-catalog\"\n",
+    "\n",
+    "# The base ApiClient is async\n",
+    "api_client = ApiClient(configuration=config)\n",
+    "\n",
+    "client = UnitycatalogFunctionClient(uc=api_client)\n",
```
Comment (BenWilson2) [inline ai/integrations/langchain/langchain_oss_example.ipynb:93]:
```suggestion
    "client = UnitycatalogFunctionClient(api_client=api_client)\n",
```

File: ai/integrations/openai/openai_oss_example.ipynb [hunk 5]
```diff
@@ -0,0 +1,394 @@
...
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "config = Configuration()\n",
+    "config.host = \"http://localhost:8080/api/2.1/unity-catalog\"\n",
+    "\n",
+    "# The base ApiClient is async\n",
+    "api_client = ApiClient(configuration=config)\n",
+    "\n",
+    "client = UnitycatalogFunctionClient(uc=api_client)\n",
```
Comment (BenWilson2) [inline ai/integrations/openai/openai_oss_example.ipynb:79]:
```suggestion
    "client = UnitycatalogFunctionClient(api_client=api_client)\n",
```

File: ai/integrations/anthropic/anthropic_oss_example.ipynb [hunk 6]
```diff
@@ -0,0 +1,302 @@
...
+   "execution_count": 3,
+   "id": "9a93a22f-4d55-428e-8576-ee11be1c009d",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "config = Configuration()\n",
+    "config.host = \"http://localhost:8080/api/2.1/unity-catalog\"\n",
+    "\n",
+    "# The base ApiClient is async\n",
+    "api_client = ApiClient(configuration=config)\n",
```
Comment (serena-ruan) [inline ai/integrations/anthropic/anthropic_oss_example.ipynb:78]: As a full executable notebook, I think we should include catalog/schema creation step

File: ai/integrations/langchain/langchain_oss_example.ipynb [hunk 7]
```diff
@@ -0,0 +1,424 @@
...
+    "    properties = material_properties[material.lower()]\n",
+    "    k = properties[\"thermal_conductivity\"]  # W/m·K\n",
+    "    A = properties[\"absorption_coefficient\"]  # Dimensionless\n",
+    "\n",
+    "    power = (k * thickness * cutting_speed) / (laser_efficiency * A)\n",
+    "\n",
+    "    return power\n",
+    "\n",
+    "\n",
+    "def calculate_cutting_speed(\n",
```
Comment (serena-ruan) [inline ai/integrations/langchain/langchain_oss_example.ipynb:182]: These two functions have a lot common steps, can we create a third function that calculates k and A and reuse it inside these two functions? Curious if it works :)

**Review by serena-ruan — APPROVED** [review]
LGTM!
Also curious if we really need two notebook examples for databricks/oss. Seems the only difference is functionClient setup stage, and we can create two separate notebooks explaining the client setup for databricks/oss and route to them.
Another thing we might want to think about is whether we should test those notebooks -- databricks ones should be able to run in our runner workspace

File: ai/integrations/langchain/langchain_oss_example.ipynb (hunk 7 above)
Comment (BenWilson2) [reply ai/integrations/langchain/langchain_oss_example.ipynb:182]: Heh... that was what I tried at first, but the number of failed calls ended up looking like something was broken (although it works with GPT4o1-preview, this is still in beta) as the tool dependency logic for multiple calls repeatedly ended up effectively giving up on calling the tools after 3-4 attempts. It seems like fully self-contained tool calls are the safest bet for Agents at this phase of their sophistication.

File: ai/integrations/anthropic/anthropic_oss_example.ipynb (hunk 6 above)
Comment (BenWilson2) [reply ai/integrations/anthropic/anthropic_oss_example.ipynb:78]: Agreed. I think I'm going to create those utility methods for create-only if not exists for catalogs and schemas within our client API, then we can (in that PR) update the notebooks and examples to use those APIs :)

**Comment by BenWilson2** [comment]
The intention for separate notebooks for OSS vs Databricks is to fully separate the experience for OSS users. The product direction for UC OSS is to not expose mention of Databricks UC at all :)

**Pull request merged**